ui/tests/integration/components/search-select-test.js

```diff
@@ -0,0 +1,164 @@
+    await render(hbs`{{search-select label="foo" inputValue=inputValue models=models onChange=onChange}}`);
+    assert.equal(component.selectedOptions.length, 1, 'there is 1 selected option');
+    await clickTrigger();
+    assert.equal(component.options.length, 2, 'shows two options');
+  });
+
+  test('it adds discarded lsit items back into select', async function(assert) {
+    const models = ['identity/entity'];
+    this.set('models', models);
+    this.set('onChange', sinon.spy());
+    this.set('inputValue', ['8']);
+    await render(hbs`{{search-select label="foo" inputValue=inputValue models=models onChange=onChange}}`);
+    assert.equal(component.selectedOptions.length, 1, 'there is 1 selected option');
+    await component.deleteButtons[0].click();
+    assert.equal(component.selectedOptions.length, 0, 'there are no selected options');
+    assert.ok(this.onChange.calledOnce);
+    await clickTrigger();
+    assert.equal(component.options.length, 3, 'shows all options');
+  });
+
+  test('it uses fallback component if endpoint 403s', async function(assert) {
+    const models = ['policy/rgp'];
+    this.set('models', models);
+    this.set('onChange', sinon.spy());
+    await render(
+      hbs`{{search-select label="foo" inputValue=inputValue models=models fallbackComponent="string-list" onChange=onChange}}`
+    );
+    assert.ok(component.hasStringList);
+  });
+
+  test('it shows no results if endpoint 404s', async function(assert) {
+    const models = ['test'];
+    this.set('models', models);
+    this.set('onChange', sinon.spy());
+    await render(
+      hbs`{{search-select label="foo" inputValue=inputValue models=models fallbackComponent="string-list" onChange=onChange}}`
+    );
+    await clickTrigger();
+    assert.equal(component.options.length, 1, 'has the disabled no results option');
+    assert.equal(component.options[0].text, 'No results found', 'text of option shows No results found');
+  });
+
+  test('it shows both name and smaller id for identity endpoints', async function(assert) {
+    const models = ['identity/entity'];
+    this.set('models', models);
+    this.set('onChange', sinon.spy());
+    await render(hbs`{{search-select label="foo" inputValue=inputValue models=models onChange=onChange}}`);
+    await clickTrigger();
+    assert.equal(component.options.length, 3, 'shows all options');
+    assert.equal(component.smallOptionIds.length, 3, 'shows the smaller id text and the name');
+  });
+
+  test('it does not show name and smaller id for non-identity endpoints', async function(assert) {
+    const models = ['policy/acl'];
+    this.set('models', models);
+    this.set('onChange', sinon.spy());
+    await render(hbs`{{search-select label="foo" inputValue=inputValue models=models onChange=onChange}}`);
+    await clickTrigger();
+    assert.equal(component.options.length, 3, 'shows all options');
+    assert.equal(component.smallOptionIds.length, 0, 'only shows the regular sized id');
+  });
+});
```

Comment:
These are _so good_ - I do have one ask though - can we add a test or two (or even assertions added in these tests) about the value of what's passed to onChange? `this.onChange.calledWithArgs` or similar - it's just that that's the interface where the component is emitting the values, so it'd be good to verify it's doing the right thing when items are added, removed, etc.